generator/src/main/java/com/scottlogic/deg/generator/generation/RegexStringGenerator.java

```diff
@@ -254,10 +254,24 @@ private String generateRandomStringInternal(
             if (!selectedTransitions.contains(nextInt)) {
                 selectedTransitions.add(nextInt);
 
-                Transition randomTransition = transitions.get(nextInt);
-                int diff = randomTransition.getMax() - randomTransition.getMin() + 1;
-                int randomOffset = diff > 0 ? random.nextInt(diff) : diff;
-                char randomChar = (char) (randomOffset + randomTransition.getMin());
+
+                /**
+                 * <p>
+                 * We have to surround this functionality in a loop checking for invalid
```

Comment:
Can you make reference to the ticket that relates to this upgrade please

Reply:
updated the comment